### PR TITLE
*: Clean up

### PR DESCRIPTION
**Pull request by zimulala**
ddl: 

- use a transaction to add a DDL job in the function of `doDDLJob` and move the function from `ddl_worker.go` to `ddl.go`

- add comments in `ddl_worker.go` 

- update comments in ddl package

model: 

- update comments

### CODE REVIEW

**Comment by disksing** [comment]
LGTM

**Comment by zimulala** [comment]
PTAL @coocood @hanfei1991 @tiancaiamao

File: ddl/ddl.go [hunk 1]
```diff
@@ -181,7 +181,7 @@ func (d *ddl) Stop() error {
 			return nil
 		}
 
-		// ddl job's owner is me, clean it so other servers can compete for it quickly.
+		// DDL job's owner is me, clean it so other servers can compete for it quickly.
```
Comment (hanfei1991) [inline ddl/ddl.go]: complete

File: ddl/ddl.go [hunk 2]
```diff
@@ -198,7 +198,7 @@ func (d *ddl) Stop() error {
 			return nil
 		}
 
-		// background job's owner is me, clean it so other servers can compete for it quickly.
+		// Background job's owner is me, clean it so other servers can compete for it quickly.
```
Comment (hanfei1991) [inline ddl/ddl.go]: ditto.

File: ddl/ddl.go [hunk 3]
```diff
@@ -223,7 +223,7 @@ func (d *ddl) start() {
 	d.wait.Add(2)
 	go d.onBackgroundWorker()
 	go d.onDDLWorker()
-	// for every start, we will send a fake job to let worker
+	// For every start, we will send a fake job to let worker
 	// check owner first and try to find whether a job exists and run.
```
Comment (hanfei1991) [inline ddl/ddl.go]: firstly

File: ddl/ddl.go [hunk 4]
```diff
@@ -372,7 +372,7 @@ func setColumnFlagWithConstraint(colMap map[string]*table.Column, v *ast.Constra
 		for _, key := range v.Keys {
 			c, ok := colMap[key.Column.Name.L]
 			if !ok {
-				// TODO: table constraint on unknown column.
+				// TODO: Table constraint on unknown column.
```
Comment (hanfei1991) [inline ddl/ddl.go]: columns

File: ddl/ddl.go [hunk 5]
```diff
@@ -383,7 +383,7 @@ func setColumnFlagWithConstraint(colMap map[string]*table.Column, v *ast.Constra
 		for i, key := range v.Keys {
 			c, ok := colMap[key.Column.Name.L]
 			if !ok {
-				// TODO: table constraint on unknown column.
+				// TODO: Table constraint on unknown column.
```
Comment (hanfei1991) [inline ddl/ddl.go]: ditto

File: ddl/ddl.go [hunk 6]
```diff
@@ -402,7 +402,7 @@ func setColumnFlagWithConstraint(colMap map[string]*table.Column, v *ast.Constra
...
```
Comment (hanfei1991) [inline ddl/ddl.go]: ditto

File: ddl/ddl.go [hunk 7]
```diff
@@ -878,8 +878,8 @@ func (d *ddl) CreateTable(ctx context.Context, ident ast.Ident, colDefs []*ast.C
 		Type:     model.ActionCreateTable,
 		Args:     []interface{}{tbInfo},
 	}
-	// Handle Table Options
 
+	// Handle Table Options.
```
Comment (hanfei1991) [inline ddl/ddl.go]: this comment makes no sense.

File: ddl/ddl.go [hunk 8]
```diff
@@ -927,7 +927,7 @@ func (d *ddl) handleTableOptions(options []*ast.TableOption, tbInfo *model.Table
 }
 
 func (d *ddl) AlterTable(ctx context.Context, ident ast.Ident, specs []*ast.AlterTableSpec) (err error) {
-	// now we only allow one schema changes at the same time.
+	// Now we only allow one schema changes at the same time.
```
Comment (hanfei1991) [inline ddl/ddl.go]: changing

File: ddl/ddl.go [hunk 9]
```diff
@@ -1007,8 +1007,8 @@ func (d *ddl) AddColumn(ctx context.Context, ti ast.Ident, spec *ast.AlterTableS
 		return ErrTooLongIdent.Gen("too long column %s", colName)
 	}
 
-	// ingore table constraints now, maybe return error later
-	// we use length(t.Cols()) as the default offset first, later we will change the
+	// Ingore table constraints now, maybe return error later.
+	// We use length(t.Cols()) as the default offset first, later we will change the
```
Comment (hanfei1991) [inline ddl/ddl.go]: firstly

File: ddl/ddl_worker.go [hunk 10]
```diff
@@ -26,94 +26,34 @@ import (
...
-		job.ID, err = t.GenGlobalID()
-		startTS = txn.StartTS()
-		return errors.Trace(err)
-	})
-	if err != nil {
-		return errors.Trace(err)
-	}
-	ddlQuery, _ := ctx.Value(context.QueryString).(string)
-	job.Query = ddlQuery
+// onDDLWorker is for async online schema change, it will try to become the owner first,
```
Comment (hanfei1991) [inline ddl/ddl_worker.go]: changing

**Comment by hanfei1991** [comment]
@queenypingcap PTAL

File: ddl/ddl.go (hunk 4 above)
Comment (hanfei1991) [inline ddl/ddl.go]: @zimulala

File: model/ddl.go [hunk 11]
```diff
@@ -78,19 +78,19 @@ type Job struct {
...
-	// every time we meet an error when running job, we will increase it.
+	// Every time we meet an error when running job, we will increase it.
 	ErrorCount int64 `json:"err_count"`
-	// the number of rows are processed.
+	// The number of rows are processed.
 	RowCount int64         `json:"row_count"`
 	Mu       sync.Mutex    `json:"-"`
 	Args     []interface{} `json:"-"`
-	// we must use json raw message for delay parsing special args.
+	// We must use json raw message for delay parsing special args.
```
Comment (hanfei1991) [inline model/ddl.go]: fot -> to

File: model/ddl.go [hunk 12]
```diff
@@ -78,19 +78,19 @@ type Job struct {
 	TableID  int64         `json:"table_id"`
 	State    JobState      `json:"state"`
 	Error    *terror.Error `json:"err"`
-	// every time we meet an error when running job, we will increase it.
+	// Every time we meet an error when running job, we will increase it.
 	ErrorCount int64 `json:"err_count"`
-	// the number of rows are processed.
+	// The number of rows are processed.
```
Comment (hanfei1991) [inline model/ddl.go]: that are

File: ddl/ddl_worker.go [hunk 13]
```diff
@@ -441,11 +384,11 @@ func toTError(err error) *terror.Error {
 		return tErr
 	}
 
-	// TODO: add the error code
+	// TODO: Add the error code.
 	return terror.ClassDDL.New(terror.CodeUnknown, err.Error())
 }
 
-// for every lease seconds, we will re-update the whole schema, so we will wait 2 * lease time
+// For every lease seconds, we will re-update the whole schema, so we will wait 2 * lease time
```
Comment (hanfei1991) [inline ddl/ddl_worker.go]: For every second in lease,

File: ddl/ddl_worker.go [hunk 14]
```diff
@@ -315,31 +286,29 @@ func (d *ddl) handleDDLJobQueue() error {
...
 			return nil
 		}
 
 		d.hookMu.Lock()
 		d.hook.OnJobUpdated(job)
 		d.hookMu.Unlock()
 
-		// here means the job enters another state (delete only, write only, public, etc...) or is cancelled.
-		// if the job is done or still running, we will wait 2 * lease time to guarantee other servers to update
+		// Here means the job enters another state (delete only, write only, public, etc...) or is cancelled.
```
Comment (hanfei1991) [inline ddl/ddl_worker.go]: is -> be

File: ddl/ddl.go (hunk 4 above)
Comment (zimulala) [inline ddl/ddl.go]: If the "unknown column" is a kind of column, I think it's OK.

Comment (hanfei1991) [inline ddl/ddl.go]: constraint ->constrains

**Comment by shenli** [comment]
LGTM

**Pull request merged**